android/src/main/java/com/google/android/react/navsdk/NavAutoModule.java

```diff
@@ -0,0 +1,537 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.react.navsdk;
+
+import android.location.Location;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.CatalystInstance;
+import com.facebook.react.bridge.NativeArray;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.UiThreadUtil;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeArray;
+import com.google.android.gms.maps.UiSettings;
+import com.google.android.gms.maps.model.CameraPosition;
+import com.google.android.gms.maps.model.Circle;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.Polygon;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.libraries.navigation.StylingOptions;
+import java.util.Map;
+
+/**
+ * This exposes a series of methods that can be called diretly from the React Native code. They have
+ * been implemented using promises as it's not recommended for them to be synchronous.
+ */
+public class NavAutoModule extends ReactContextBaseJavaModule implements INavigationAutoCallback {
```

Comment:
Nit: Is there a way to consolidate the methods here with the other NavModule?

Reply:
Not an easy way as NavAutoModule is implementing same functionality as NavViewModule. And as normal nav views can be controlled directly with RN view manager, the functionality differs.

NavModule handles the navigation; and this functionality is separate from view controls.